plugins/GSdx/GSTextureCache.cpp

```diff
@@ -268,25 +270,20 @@ GSTextureCache::Source* GSTextureCache::LookupSource(const GIFRegTEX0& TEX0, con
 				uint32 t_psm = (t->m_dirty_alpha) ? t->m_TEX0.PSM & ~0x1 : t->m_TEX0.PSM;
 
 				if (GSUtil::HasSharedBits(bp, psm, t->m_TEX0.TBP0, t_psm)) {
-					if (!s_IS_OPENGL && (psm == PSM_PSMT8)) {
-						// OpenGL can convert the texture directly in the GPU. Not sure we want to keep this
-						// code for DX. It fixes effect but it is slow (MGS3)
-
-						// It is a complex to convert the code in shader. As a reference, let's do it on the CPU, it will
-						// be slow but
-						// 1/ it just works :)
-						// 2/ even with upscaling
-						// 3/ for both DX and OpenGL
-
-						// Gregory: to avoid a massive slow down for nothing, let's only enable
-						// this code when CRC is below the FULL level
-						if (m_crc_hack_level < CRCHackLevel::Full)
-							Read(t, t->m_valid);
-						else
-							dst = t;
-					} else {
+					// It is a complex to convert the code in shader. As a reference, let's do it on the CPU, it will be slow but
+					// 1/ it just works :)
+					// 2/ even with upscaling
+					// 3/ for both DX and OpenGL
+					if (m_cpu_8bit_fb_conversion && psm == PSM_PSMT8)
```

Comment:
Add also the 4 bits format `PSM_PSMT4`. I don't know if some game use it but there is no shader to convert this format. This way it would help to test bad graphics.

Reply:
Seems weird to have the hack named 8bit after this change. CPU framebuffer conversion ?

Reply:
well it isn't important. Otherwise rename 8bits to index or palette format. Note the other 8 bit option handles also both format.

Reply:
For the option in the gui maybe
CPU Texture conversion
To keep things simple.